src/views/ManageList.jsx

```diff
@@ -18,17 +18,50 @@ export function ManageList({ listPath, userId }) {
 		theme: 'light',
 	};
 
+	// Fetch existing items from the list
+	const existingItems = useShoppingListData(listPath);
+
 	const handleItemSubmit = async (e) => {
 		e.preventDefault();
+		// checks if the user submits an empty item
+		if (!itemName.trim()) {
+			toast.error('Item cannot be empty');
+			return;
+		}
+
+		// Normalize the entered item name (lowercase and remove punctuation)
+		const normalizedItemName = itemName
+			.trim()
+			.toLowerCase()
+			.replace(/[^\w\s]/gi, '');
+
+		// Check if the entered item name already exists in the list with normalized casing and punctuation
+		const itemExists = existingItems.some(
+			(item) =>
+				item.name.toLowerCase().replace(/[^\w\s]/gi, '') === normalizedItemName,
+		);
+
+		if (itemExists) {
+			// If the item already exists, display an error message and exit the function
+			toast.error('Item already exists in the list');
+			return;
+		}
+
+		// Add the new item to the database
+
 		try {
 			await addItem(listPath, {
-				itemName,
+				itemName: normalizedItemName, // Pass normalized item name
+				originalItemName: itemName, // Pass original item name
```

Comment:
the field `originalName` is unnecessary. If the item does not exist, we should save the name as entered by the user

Reply:
The acceptance criteria state that we need to store the `originalName` first. That's why I made sure to define it before proceeding with normalization.